src/docker/composebuilder/service_endpoints.go

```diff
@@ -46,13 +54,17 @@ func (s *ServiceEndpoints) GetPortMappings() []string {
 
 // GetEndpointMappings returns a map from env var name to env var value of a service endpoint
 func (s *ServiceEndpoints) GetEndpointMappings() map[string]string {
-	if s.ContainerPort == "" {
-		return map[string]string{}
-	}
 	switch s.ServiceConfig.Type {
 	case "public":
 		externalKey := fmt.Sprintf("%s_EXTERNAL_ORIGIN", toConstantCase(s.ServiceRole))
-		externalValue := fmt.Sprintf("http://localhost:%s", s.HostPort)
+		var externalValue string
+		if s.BuildMode.Type == BuildModeTypeLocal {
+			if s.HostPort != "" {
+				externalValue = fmt.Sprintf("http://localhost:%s", s.HostPort)
+			}
+		} else {
+			externalValue = fmt.Sprintf("https://%s.%s.com", s.ServiceRole, s.AppContext.Config.Production.URL)
```

Comment:
Did you check with @alexdavid about this? I believe he will be a user of this so wanted to verify that this is what it should be.

Reply:
Just spoke with Alex, we should be using the `ServiceConfig.Production.URL` endpoint, which I just switched it to.